File: resto/resto_sopwer/report/daily_sales_report/daily_sales_report.js
// Copyright (c) 2026, PT Sopwer Teknologi Indonesia and contributors
// For license information, please see license.txt

frappe.query_reports["Daily Sales Report"] = {
    filters: [
        {
            fieldname: "from_date",
            label: __("From Date"),
            fieldtype: "Date",
            default: frappe.datetime.month_start(),
            reqd: 1,
        },
        {
            fieldname: "to_date",
            label: __("To Date"),
            fieldtype: "Date",
            default: frappe.datetime.now_date(),
            reqd: 1,
        },
        {
            fieldname: "branch",
            label: __("Branch"),
            fieldtype: "Link",
            options: "Branch",
            default: frappe.defaults.get_user_default("branch") || "",
        },
    ],

    onload(report) {
        const setRange = (from, to) => {
            report.set_filter_value("from_date", from);
            report.set_filter_value("to_date", to);
        };

        report.page.add_inner_button(__("Hari Ini"), () => {
            const today = frappe.datetime.now_date();
            setRange(today, today);
        });

        report.page.add_inner_button(__("Minggu Ini"), () => {
            setRange(frappe.datetime.week_start(), frappe.datetime.now_date());
        });

        report.page.add_inner_button(__("Bulan Ini"), () => {
            setRange(frappe.datetime.month_start(), frappe.datetime.now_date());
        });

        report.page.add_inner_button(__("Print Lengkap (PDF)"), () => {
            const filters = report.get_values();
            if (!filters.from_date || !filters.to_date) {
                frappe.msgprint(__("Pilih From Date dan To Date terlebih dahulu"));
                return;
            }
            if (!filters.branch) {
                frappe.msgprint(__("Pilih Branch terlebih dahulu untuk Print Lengkap"));
                return;
            }
            const url =
                "/api/method/resto.api.print_daily_sales_full_pdf" +
                "?from_date=" + encodeURIComponent(filters.from_date) +
                "&to_date=" + encodeURIComponent(filters.to_date) +
                "&branch=" + encodeURIComponent(filters.branch);
            window.open(url, "_blank");
        });
    },
};

File: resto/resto_sopwer/report/daily_sales_report/daily_sales_report.py
# Copyright (c) 2026, PT Sopwer Teknologi Indonesia and contributors
# For license information, please see license.txt

from resto.services.reporting_service import ReportingService


def execute(filters=None):
    filters = filters or {}
    data = ReportingService().get_daily_sales_summary(
        from_date=filters.get("from_date"),
        to_date=filters.get("to_date"),
        branch=filters.get("branch"),
    )
    return get_columns(), data


def get_columns():
    return [
        {"label": "Tanggal", "fieldname": "posting_date", "fieldtype": "Date", "width": 110},
        {"label": "Cabang", "fieldname": "branch", "fieldtype": "Link", "options": "Branch", "width": 140},
        {"label": "Pax", "fieldname": "total_pax", "fieldtype": "Int", "width": 80},
        {"label": "Bill", "fieldname": "total_bill", "fieldtype": "Int", "width": 80},
        {"label": "Sub Total", "fieldname": "sub_total", "fieldtype": "Currency", "width": 140},
        {"label": "Discount", "fieldname": "discount", "fieldtype": "Currency", "width": 130},
        {"label": "Tax", "fieldname": "tax", "fieldtype": "Currency", "width": 120},
        {"label": "Grand Total", "fieldname": "grand_total", "fieldtype": "Currency", "width": 160},
        {"label": "Void Bill", "fieldname": "void_bill", "fieldtype": "Int", "width": 90},
        {"label": "Void Amount", "fieldname": "void_amount", "fieldtype": "Currency", "width": 140},
        {"label": "Draft Bill", "fieldname": "draft_bill", "fieldtype": "Int", "width": 90},
        {"label": "Draft Amount", "fieldname": "draft_amount", "fieldtype": "Currency", "width": 140},
    ]
